include/alpaka/dev/DevOacc.hpp

```diff
@@ -0,0 +1,343 @@
+                            ++it;
+                        }
+                        else
+                        {
+                            it = m_queues.erase(it);
+                        }
+                    }
+                    return vspQueues;
+                }
+
+                //-----------------------------------------------------------------------------
+                //! Registers the given queue on this device.
+                //! NOTE: Every queue has to be registered for correct functionality of device wait operations!
+                ALPAKA_FN_HOST auto registerQueue(std::shared_ptr<IGenericThreadsQueue<DevOacc>> spQueue)
+                -> void
+                {
+                    std::lock_guard<std::mutex> lk(m_Mutex);
+
+                    // Register this queue on the device.
+                    m_queues.push_back(spQueue);
+                }
+
+                int iDevice() const {return m_iDevice;}
+                acc_device_t deviceType() const {return m_deviceType;}
+
+            private:
+                std::mutex mutable m_Mutex;
+                std::vector<std::weak_ptr<IGenericThreadsQueue<DevOacc>>> mutable m_queues;
+                acc_device_t m_deviceType;
+                int m_iDevice;
+            };
+        }
+    }
+    //#############################################################################
+    //! The Oacc device handle.
+    class DevOacc :
+        public concepts::Implements<ConceptCurrentThreadWaitFor, DevOacc>,
+        public concepts::Implements<ConceptDev, DevOacc>
+    {
+        friend struct traits::GetDevByIdx<PltfOacc>;
+
+    protected:
+        //-----------------------------------------------------------------------------
+        DevOacc(int iDevice) :
+            m_spDevOaccImpl(std::make_shared<oacc::detail::DevOaccImpl>(iDevice))
+        {}
+    public:
+        //-----------------------------------------------------------------------------
+        DevOacc(DevOacc const &) = default;
+        //-----------------------------------------------------------------------------
+        DevOacc(DevOacc &&) = default;
+        //-----------------------------------------------------------------------------
+        auto operator=(DevOacc const &) -> DevOacc & = default;
+        //-----------------------------------------------------------------------------
+        auto operator=(DevOacc &&) -> DevOacc & = default;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator==(DevOacc const & rhs) const
+        -> bool
+        {
+            return m_spDevOaccImpl->iDevice() == rhs.m_spDevOaccImpl->iDevice();
+        }
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator!=(DevOacc const & rhs) const
+        -> bool
+        {
+            return !((*this) == rhs);
+        }
```

Comment:
Prefer defining relational operators as hidden friends.

Reply:
I think it is done this way for consistency with the other devices which do the same. We should not diverge here but do it consistently. As this is already a large PR, changing the other impls should not be added to this PR so the change should an independent PR.

Reply:
I created issue #1201 for this.